checkov/common/runners/object_runner.py

```diff
@@ -336,8 +336,12 @@ def _get_jobs(self, definition: dict[str, Any]) -> dict[int, str]:
                     end_line: int = job_instance.get(END_LINE, -1)
                     end_line_to_job_name_dict[end_line] = job_name
 
-                    steps = [step for step in job_instance.get('steps', []) or [] if step]
-                    if steps:
-                        for step in steps:
-                            end_line_to_job_name_dict[step.get(END_LINE)] = job_name
+                    steps: list[dict[str, Any]] = [step for step in job_instance.get('steps', []) or [] if step]
+                    if not steps:
+                        continue
+
+                    for step in steps:
+                        if not isinstance(step, dict):
+                            continue
+                        end_line_to_job_name_dict[step.get(END_LINE)] = job_name  # type: ignore
```

Comment:
does step.get(END_LINE) will surely be in the end_line_to_job_name_dict?

Reply:
The step won't necessarily be there, but the encapsulating job will be

Reply:
so, if `step.get(END_LINE)` can be `None` then you will get a `KeyError` here

Reply:
I think best is to check if `END_LINE` is in `step` beforehand, just be sure. And also add the error code to the `type:ignore` if it is really needed.

Reply:
@gruebel Fixed
The `type:ignore` is needed because the values type of `step` is `Any` while the keys type of `end_line_to_job_name_dict` are `int`